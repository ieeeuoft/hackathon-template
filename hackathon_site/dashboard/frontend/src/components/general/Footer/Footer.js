import React from "react";
import Typography from "@material-ui/core/Typography";
import Container from "@material-ui/core/Container";
import styles from "./Footer.module.scss";

const Footer = () => {
    let fakeDate = new Date();
    const yearRange =
        fakeDate.getFullYear() === 2020 ? "" : " - " + fakeDate.getFullYear();
    const outputMessage =
        "Â© 2020" +
        yearRange +
        " Built by the web team at MakeUofT (University of Toronto)";

    return (
        <Container maxWidth={false} disableGutters={true}>
            <footer className={styles.footer}>
                <Typography variant="caption" display="block" gutterBottom>
                    Hardware Signout
                </Typography>
                <Typography variant="caption" display="block" gutterBottom>
                    {outputMessage}
                </Typography>
            </footer>
        </Container>
    );
};

export default Footer;
