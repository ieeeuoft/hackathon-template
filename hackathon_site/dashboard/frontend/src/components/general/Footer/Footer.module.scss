@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.footer {
    text-align: center;
    background-color: color(footer);
    color: color(white);
    padding: 10px;
    width: 100%;
}
