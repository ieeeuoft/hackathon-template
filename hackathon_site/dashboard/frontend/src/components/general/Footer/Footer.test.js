import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Footer from "./Footer";

describe("Footer", () => {
    test("renders", () => {
        const fakeYear = 2020;
        global.Date.prototype.getFullYear = jest.fn(() => fakeYear);

        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("date", () => {
        const fakeYear = 2021;
        global.Date.prototype.getFullYear = jest.fn(() => fakeYear);

        const { getByText } = render(<Footer />);
        const footerText = getByText(/Built/, { exact: false });

        expect(footerText).toHaveTextContent(
            `Â© 2020 - ${fakeYear} Built by the web team at MakeUofT (University of Toronto)`
        );
    });
});
