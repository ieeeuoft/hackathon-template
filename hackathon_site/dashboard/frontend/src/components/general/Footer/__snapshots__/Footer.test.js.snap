// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-disableGutters"
  >
    <footer
      class="footer"
    >
      <span
        class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom MuiTypography-displayBlock"
      >
        Hardware Signout
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom MuiTypography-displayBlock"
      >
        Â© 2020 Built by the web team at MakeUofT (University of Toronto)
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
